rootchain/sampool/pool_test.go

```diff
@@ -0,0 +1,356 @@
+
+			assert.ErrorIs(t,
+				pool.AddMessage(
+					rootchain.SAM{
+						Event: rootchain.Event{Index: 3},
+					},
+				),
+				ErrStaleMessage,
+			)
+		},
+	)
+
+	t.Run(
+		"message added",
+		func(t *testing.T) {
+			t.Parallel()
+
+			verifier := mockVerifier{
+				verifyHash:      func(rootchain.SAM) error { return nil },
+				verifySignature: func(rootchain.SAM) error { return nil },
+			}
+
+			pool := New(verifier)
+
+			msg := rootchain.SAM{
+				Hash: types.Hash{111},
+				Event: rootchain.Event{
+					Index: 3,
+				},
+			}
+
+			assert.NoError(t, pool.AddMessage(msg))
+
+			bucket, ok := pool.messages[msg.Index]
+			assert.True(t, ok)
+			assert.NotNil(t, bucket)
+
+			set, ok := bucket[msg.Hash]
+			assert.True(t, ok)
+			assert.NotNil(t, set)
+
+			messages := set.get()
+			assert.NotNil(t, messages)
+			assert.Len(t, messages, 1)
+		},
+	)
+
+	t.Run(
+		"no duplicate message added",
+		func(t *testing.T) {
+			t.Parallel()
+
+			verifier := mockVerifier{
+				verifyHash:      func(rootchain.SAM) error { return nil },
+				verifySignature: func(rootchain.SAM) error { return nil },
+			}
+
+			pool := New(verifier)
+
+			msg := rootchain.SAM{
+				Hash:      types.Hash{111},
+				Signature: []byte("signature"),
+				Event: rootchain.Event{
+					Index: 3,
+				},
+			}
+
+			assert.NoError(t, pool.AddMessage(msg))
+
+			bucket, ok := pool.messages[msg.Index]
+			assert.True(t, ok)
+			assert.NotNil(t, bucket)
+
+			set, ok := bucket[msg.Hash]
+			assert.True(t, ok)
+			assert.NotNil(t, set)
+
+			messages := set.get()
+			assert.NotNil(t, messages)
+			assert.Len(t, messages, 1)
+
+			//	add the message again
+			assert.NoError(t, pool.AddMessage(msg))
+
+			//	num of messages is still 1
+			set = pool.messages[msg.Index][msg.Hash]
+			assert.Len(t, set.get(), 1)
+		},
+	)
+}
+
+func TestSAMPool_Prune(t *testing.T) {
+	t.Parallel()
+
+	t.Run(
+		"Prune removes message",
+		func(t *testing.T) {
+			t.Parallel()
+
+			verifier := mockVerifier{
+				verifyHash:      func(rootchain.SAM) error { return nil },
+				verifySignature: func(rootchain.SAM) error { return nil },
+			}
+
+			pool := New(verifier)
+
+			msg := rootchain.SAM{
+				Hash: types.Hash{111},
+				Event: rootchain.Event{
+					Index: 3,
+				},
+			}
+
+			assert.NoError(t, pool.AddMessage(msg))
+
+			_, ok := pool.messages[msg.Index]
+			assert.True(t, ok)
+
+			pool.Prune(5)
+
+			_, ok = pool.messages[msg.Index]
+			assert.False(t, ok)
+		},
+	)
+
+	t.Run(
+		"Prune removes no message",
+		func(t *testing.T) {
+			t.Parallel()
+
+			verifier := mockVerifier{
+				verifyHash:      func(rootchain.SAM) error { return nil },
+				verifySignature: func(rootchain.SAM) error { return nil },
+			}
+
+			pool := New(verifier)
+
+			msg := rootchain.SAM{
+				Hash: types.Hash{111},
+				Event: rootchain.Event{
+					Index: 10,
+				},
+			}
+
+			assert.NoError(t, pool.AddMessage(msg))
+
+			_, ok := pool.messages[msg.Index]
+			assert.True(t, ok)
```

Comment:
You can use `assert.Contains`

Reply:
[Fixed](https://github.com/0xPolygon/polygon-edge/pull/718/commits/fd6d964419635c317abbf5cf19a42827d3a89974)